pkg/rancher-desktop/config/settings.ts

```diff
@@ -186,14 +192,35 @@ export async function clear() {
 export function getUpdatableNode(cfg: Settings, fqFieldAccessor: string): [Record<string, any>, string] | null {
   const optionParts = fqFieldAccessor.split('.');
   const finalOptionPart = optionParts.pop() ?? '';
-  let currentConfig: Record<string, any> = cfg;
+  const currentConfig = optionParts.length === 0 ? cfg : _.get(cfg, optionParts.join('.'));
+
+  return (finalOptionPart in (currentConfig || {})) ? [currentConfig, finalOptionPart] : null;
+}
+
+export function getObjectRepresentation(fqFieldAccessor: string, finalValue: boolean|number|string): RecursivePartial<Settings> {
+  if (!fqFieldAccessor) {
```

Comment:
Nope, because `getObjectRepresentation` also does some error-checking.

`getObjectRepresentation('application.', 4)` throws an error, while
`_.set({}, 'application.', 4)` returns `{ application: { '': 4 } }`